%In light of this discussion I will also need to change the discussion in the
% next chapter
%somehow need to include the word constructive more, that's key
Life is produced and sustained through the organisation of entities at various
levels (\eg molecules, membranes) that interact to construct other autonomous
entities (such as cells). Organisation and structure are therefore fundamental
and defining properties of all living systems. Some go as far as to say that
(self-)organisation and the resulting self-maintenance are the only common and
defining characteristics of living organisms and that they are more important
than even natural selection in shaping the forms of life that we see
\citep[sometimes referred to as structuralism, for
example;][]{thompson1942growth, kauffman1992origins}. Even if we do not take
such an extreme view, it is still true that if we want to understand life we
have to understand the organisation that sustains it and the links between
processes at all scales defined by this organisation -- from genes to organisms
and ecosystems. Organisation here means both spatial organisation, for example
cells in a tissue, but also relations or interactions between processes that
give rise to other higher-level phenomena or between processes and the
environment.

% other less fundamental question could possibly be answered but at the core
% we need this
This view of organisation as a fundamental and defining principle of biology
makes, in my opinion, the subject distinct from other natural sciences, like
physics for example and precludes a natural mode of enquiry, which is
abstraction. The nature of the questions is different. How can we abstract the
details of an organism if those details are exactly what makes the organism
alive and in fact what we are trying to understand? Since abstraction or at
least complete abstraction is not available, in order to answer biological
questions we have to take a more holistic and multi-scale view of the natural
systems of interest.

%reductionist vs system (organisation)
Despite the importance of organisation in understanding life, a common focus in
biology has been on understanding individual mechanisms at the molecular
level. The reductionist view of life is that if the cell is the main unit of
life then if we understand all the processes inside the cell we will understand
life, as everything else follows.

There were, however, even if on the fringes or often completely outside
mainstream biology, few historical (and perhaps independent) threads of work
that recognised the importance of organisation and consequently of the systems
view. C.H. Waddington, for example, talked about the importance of the
interaction of parts in developmental systems \citep{allen_evolution_1977},
perhaps inspired by the metaphysics of Alfred North Whitehead \citep[same
book;][Autobiographical note]{allen_evolution_1977}. While the polarity between
constituents of systems and the importance of their organisation (systems view)
might have been a topic of discussion in the field at the time
\citep{waddington_nature_1961}, \citet{rashevsky_topology_1954} explained the
need for a theoretical study of organisation. He called this organisation the
`relational' aspects of biology that needs to be addressed in theoretical
studies to complement usual approaches that follow only the metric aspects of
physical systems. This is the usual abstraction in Physics where systems are
viewed only by their quantifiable properties (captured by numeric
variables). \citet{rosen_relational_1958} developed such a theory of
organisation, based on the (meta-)mathematics of category theory, for the
relational/organisational aspects of biology and was a proponent of a systems
view \citep{rosen1991life}. Again, thinking about questions on the origin of
life, \citet{varela_autopoiesis:_1974} developed the concept of `autopoieisis'
(self creation through organisation) as a necessary condition for autonomous
living entities. Similar questions regarding the origin of macromolecules in
cells lead to a study of self-organisation by
\citet{eigen_selforganization_1971}. Following this \citet{fontana_what_1994}
working with Peter Schuster (Eigen's student) highlighted the importance of
organisation as the defining characteristic of living systems and revisited the
problem of developing a theory of organisation as a basis for biological
understanding. For this theory they borrowed from computer science
\citep[$\lambda$-calculus in particular;][]{fontana_barrier_1996}. Finally, the
systems view entered the mainstream through the \emph{systems biology} movement
\citep{kitano2002systems} as a way to consolidate increasingly diverse
experimental molecular biology datasets.

% need for modelling
As the systems we are trying to understand become more complex, which is
particularly the case when we take the systems view, intuitive thinking and
mental models of the processes are not adequate tools to aid our
understanding. A useful tool in these cases is formal models where the systems
are represented using some formal language based on a mathematical
interpretation, either directly or indirectly. An interesting view is that of
formal models as machines that turn the assumptions that we put into them to
conclusions \citep{gunawardena_models_2014}. The unique advantage of formal
models compared to informal ones (mental models) in biology is that this machine
is guaranteed to function reliably therefore providing a good tool for testing our
hypotheses (assumptions) just like an experiment would. Mental models however do
not provide this guarantee and we therefore do not know if the fault is in our
assumption or in the animation of the model in our heads.

Historically, it is no surprise that formal models have been used in systems
models, for example for morphogenesis \citep{turing_chemical_1952} and neuronal
activity \citep{hodgkin_quantitative_1952} although in some cases good
abstraction theories have been used in evolutionary genetics \citep[by
abstracting away the organism;][]{dobzhansky1964genetics} and in the small scale
for chemical kinetics \citep{michaelis1913kinetics, gunawardena_lessons_2012},
\citep[see also;][]{gunawardena2013biology}. With the movement of systems
biology as the intellectual successor to the system and organisation theorists
of the 20th century we have seen an increase in mathematical models of cellular
processes, from metabolism to the circadian clock.

% multi-scale models
% modelling with explicity representation of processes at multiple levels of
% organisation to understand the links between scales and how life or particular
% phenomena in life emerge through this, which cannot be explained only as a
% sum of their parts.
% also cite 
Taking the systems and modelling view to the extreme and remembering the
fundamental goal of understanding living organisms through their organisation,
we can take a step further and expand our systems view from the cell to larger
scales, for example to the organism and beyond. These models that reach beyond
organisation on a single scale are usually called \emph{multi-scale models} and
they are more close to the historical spirit of understanding system
organisation to understand life, as we have seen above. For small
micro-organisms, comprehensive models link the metabolic and molecular level
with the cellular \citep{karr_whole-cell_2012} and population growth scales
\citep{weise_mechanistic_2015}. Work on more complex organisms has focused more
narrowly (Virtual liver, \citet{holzhutter2012virtual}; Virtual heart;
\citet{noble_modeling_2002}; Bone system, \citet{paoletti_multilevel_2012})
although there are ambitious collaborative project aiming at whole organisms
\citep[virtual rat, virtual human][]{beard_multiscale_2012, kohl_systems_2009}.

%multi-scale models in plants
%plants
Plant science research is concentrated on the laboratory model species
\emph{Arabibopsis thaliana}, which offers an opportunity for broad understanding
that includes mechanistic models \citep{chew_mathematical_2014,
  voss_modelling_2014}. This multi-scale view starting from a molecular
mechanism and metabolism is realised explicitly in the Framework Model that
represents vegetative growth starting from a specific well-understood
molecular mechanism, the circadian clock \citep[FMv1]{chew_multiscale_2014}. On
the other end of the scales spectrum there is modelling work for plants in the
crop and ecology world where the organism is usually absent. While this work
usually is concerned with more crop or ecologically relevant species, recently
Arabidopsis is starting to appear in crop and ecology research in attempts to
bring the organism back into the equation and provide insight into relevant
mechanisms that affect ecological or crop phenotypes especially with respect to
the interaction with the environment.

The work of this thesis is situated exactly in this space, in the intersection
and modelling space between plant science and crop or ecological
research. We are particularly interested in linking multi-scale plant models
that consider growth through environmental interaction with ecological models
that consider ecosystems in the natural environment.  Apart from the scientific
challenges these multi-scale models present, in plants or elsewhere, there are
also some challenges regarding the technical language used for their
representation.

\section{Aims}
\label{sec:aims}
Therefore, the \emph{aims} of this thesis are:
\begin{itemize}
\item Develop a suitable language to address the problem of naturally and
  succinctly representing multi-scale plant models that go beyond the organism
  and act in a dynamic natural environment.
\item Use the above language to create such multi-scale plant models from the
  organism to the ecosystem.
\item Employ the models for answering questions in ecological models using the
  mechanistic understanding that comes from including a model of the organism in
  ecological studies.
\end{itemize}

In the rest of this chapter I will first give a simplified overview of modelling
in the plant domain starting from plant science and going to ecology and
crops. My objective here is to locate and motivate the challenges for modelling
languages, rather than a comprehensive survey of the state of the art in plant
models. I will then overview the language question in multi-scale models that
relates to the organisation view of living organisms we noted above. Finally, I
give the contributions of this thesis in reference to the outlined aims.

\section{From plants to crops and ecosystems}
\label{sec:resCircle}
Much contemporary work in plant biology research has been focused on
understanding molecular mechanisms at the foundation of important plant
processes like photosynthesis. Unlike mammalian systems, however, the effect of
environmental conditions is usually taken into account since plants are highly
dependent on those. The importance for crop traits also led to attempts at
linking molecular knowledge to organism traits, like growth. We can think of
these kinds of questions of how a particular phenotypic trait arises from a
molecular mechanism as the \emph{how} questions (top part of cycle in
Figure~\ref{fig:circle}).

\begin{figure}[tb]
  \centering
  \includegraphics[width=0.8\textwidth]{figures/circle.eps}
  \caption{Plant biology research is usually concerned with the top part of the
    cycle and the 'how' of organisms. Evolution and ecology are concerned with
    the why although they usually use a limited view of the 'how' to make
    hypotheses about the 'why' \citep[adapted
    from][]{millar_intracellular_2016}.}
  \label{fig:circle}
\end{figure}

In the fields of ecology and evolution on the other hand, the question can also
address more on they \emph{why} organisms evolved the way they have (bottom part
of cycle in Figure~\ref{fig:circle}). This is usually done by observing genetic
variation and with some knowledge of the how this affect organism behaviour make
evolutionary hypotheses on the why.

Multi-scale plant models that go beyond the organism can be seen as tools for
understanding (evolutionary ecology) and tools for engineering (crop
science). Next, we start with plant models with an emphasis on the Framework
Model of \citet{chew_multiscale_2014} and go to an overview of crops and
evolutionary ecology models.

\subsection{Multi-scale plant models}
%main point here to give an overview, a feeling for plant models all the way to
%the multiscale FM, that represents the constructive organisation of the
%organism, then limitations on scaling beyond the organism to the evolutionary
%ecology level
Modelling in plant biology is usually concerned with individual mechanisms, for
example photosynthesis \citep{farquhar_biochemical_1980}, leaf stomata
conductance \citep{tardieu1993integration}. What perhaps makes it distinct from
modelling work in other species is the importance of the environment and the
attempt to link these molecular processes to organism traits directly important
for agriculture. The time keeping mechanism of the circadian clock is
particularly important for the synchronisation with the outside environment and
has been studied with the use of models both in \textit{Arabidopsis Thaliana}
\citep{locke2005extension} and in cyanobacteria \citep{miyoshi2007mathematical,
  hertel2013revealing}. Photosynthesis models have been linked to sucrose
synthesis \citep{zhu_e-photosynthesis:_2013}, the circadian clock to flowering
time \citep{salazar_prediction_2009}, and leaf-stomata conductance to leaf
growth \citep{tardieu2015modelling}.

Unlike other organisms most of the development of plants happens throughout
their lifecycle and a lot of work was based on descriptive models of their
structural growth \citep{mundermann_quantitative_2005}. As the role of molecular
mechanisms into development became clearer there is more multi-scale work
linking molecular mechanisms to development, for example there are models of the
distribution of the hormone auxin along the plant structure, which is known to
act as a morphogen \citep{prusinkiewicz2009control,
  jonsson2006auxin}. Functional-structural models also try to link molecular
mechanisms, usually abstract views of metabolism, to structural growth at the
organ level \citep{christophe_model-based_2008}.

Plant development models usually ignore most of the underlying biochemistry of
the organism except for a very narrow description of directly relevant molecular
mechanisms. On the other hand models of molecular mechanisms, while sometimes
linked to the organism, they usually ignore the constructive organisation of the
plant. The question is can we construct the organism representing its
constructive organisation and underlying mechanisms? We next overview one
attempt at a digital organism, the Framework Model, which is inspired by the
functional-structural models.

% Then FM more multi-scale
% ->
\subsubsection*{Framework Model}
\label{subsec:fm}
The Framework Model represents vegetative growth of Arabidopsis in lab conditions
\citep{chew_multiscale_2014}, starting from four independent models that
represent photosynthesis and carbon storage \citep{rasse_leaf_2006}, plant
structure and carbon partitioning among organs
\citep{christophe_model-based_2008}, flowering phenology
\citep{chew_augmented_2012} and the circadian clock gene circuit and its output
to photoperiodic flowering \citep{salazar_prediction_2009}. Later updates
focussed on plant phenotypes controlled by the clock, such as tissue elongation
and starch metabolism \citep[FMv2;][]{chew_linking_2017}, or temperature and
organ-specific inputs to flowering
\citep{kinmonth-schultz_mechanistic_2018}. The Framework Models align with
community efforts to link understanding of crop plant processes at multiple
scales, for benefits in agriculture \citep{wu_connecting_2016, zhu_plants_2016}.

Among the limitations of the Framework Models, growth was limited to the
vegetative stage, ending upon flower induction. Without reproduction, the models
had no seed yield or link to evolutionary fitness. Without seed dormancy, they
lacked a major determinant of Arabidopsis life history in the field. Their
representation of the circadian clock was also unnecessarily detailed for many
studies outside chronobiology. These limitations mean that going beyond the
organism to the population level to answer ecological questions is not
practically possible without extra effort.

\subsection{Crop modelling}
%main point here is to give a feeling of how crop models are used, usually, for
%a  fundamentally different reason -- for engineering
%do we have any examples of this? I mean of actually doing this, genetic engineering
Crop models use many of the same ideas that we have seen in multi-scale plant
models but they have a different more engineering oriented goal in mind instead
of the fundamental biology goal of understanding. Usually the motivation is
understanding the effect of interventions to either plant genetics or crop
management on plant performance, which is usually growth related.

% traditional crop models
Crop models often regard the entire crop as one big plant. The challenges
relate exactly to this assumption. For example, computing light interception of
the canopy has to take into account the geometry and assumed configuration and
positioning of plants in the field. The constructive organisation of plants is
absent and plant processes, like photosynthesis, are scaled to the entire crop
\citep{monteith_light_1965}. Genetics or molecular mechanisms are represented,
if at all, in a phenomenological way. For example, a mutant might have a
different set of model parameters and so on \citep{parent_can_2014}.

Since the goal in these models is prediction, as long as they come
reasonably close to the behaviour of the crop under different environmental
conditions they are considered to be adequate representations. However, because
of their nature it becomes difficult to capture the responses of the systems
under multiple environmental inputs.

\subsubsection*{A flows-based view of a plant}
%Main point is to introduce the concept of this reservoir flow view, which is
%important for later
Other approaches start from a whole-plant model and then try to extrapolate to
the entire crop. The usual representation is of the plant as multiple reservoirs
of nutrients with flows between them \citep{france_mathematical_1984}. Growth,
which is the main goal usually, is calculated as a function of the levels in
these nutrient pools (see Figure~\ref{fig:plantFlow}).

\begin{figure}[tb]
\centering  
\includegraphics[scale=0.8]{figures/flowPlant.eps}
\caption{A generic plant model with the flows-based view of a plant
  \citep[adapted from][]{france_mathematical_1984}. The plant is represented by several state
  variables representing the levels of various nutrients in different parts of
  the plant. The dynamics are then seen as flows between the pools. The pools
  could also represent entities outside the system, like the soil, for
  example. The variables represent, structural mass ($X$), nitrogen ($N$),
  carbon ($C$), and water ($Wa$). The output is usually the biomass of each part
  of the plant and especially that of fruits or seeds, which are important in
  agriculture.}
  \label{fig:plantFlow}
\end{figure}

This view has later found its way into more traditional plant models and it is
the metabolism level representation found in the Framework Model, for example,
although the flows are calculated more mechanistically and the pools are at the
organ level reflecting the explicit representation of plant organisation. Again,
in this whole plant flow-based models, the constructive organisation of the
plant is usually absent preventing fundamental questions. However, as long as
the goal is prediction for engineering, then often this kind of representation
is adequate.


\subsubsection*{Multi-scale models}
The multi-scale systems modelling approach has also been proposed in crop
modelling in attempts to engineer crop traits starting from genetics or from
genomes \citep{welch_merging_2005, yin_applying_nodate, yin_modelling_2010,
  parent_can_2014, wu_connecting_2016, chenu_integrating_2018}, where simpler
models have demonstrated both the potential of crop modelling in general and the
significant demands of detailed models for empirical data that varies in
availability \citep{hammer_models_2006,asseng_uncertainty_2013}.


\subsection{Ecology and evolution}
% Main point is to introduce phenology models that while they make evolutionary
% assumptions, they rarely include fitness or genetics (genetic variation) +
% usually developmental events in isolation -- organisms is absent
%
% on the other hand genetic studies on genetic variation dont have any modelling
The lifecycle of Arabidopis plants can be divided into three major
developmental stages where the transition between them is marked by
developmental events. Starting out as a seed the plant germinates transitioning
to the vegetative stage after which it bolts going to the reproductive stage
that ends with the dropping of the new seeds (Figure~\ref{fig:plantLife}).

In natural settings the timing of these events throughout the year
(\emph{phenology}) determines the environment that the plant is exposed to
during growth. Therefore phenology is a major determinant of plant fitness. If
growth coincides, for example, with unfavourable weather conditions the plant
might not have enough resources to put into making seeds or might not even
survive to reproduction.

\begin{figure}[tb]
\centering
\includegraphics[width=0.8\textwidth]{figures/phenology.eps}
  \caption{Left: Plant developmental stages Right: Two example timings of the
    growth period in the year (phenology). The vegetative stage is shown in
    green and the reproductive in red.}
\label{fig:plantLife}
\end{figure}

Yet species have a natural geographic range that spans a wide range of climatic
conditions. For example, Arabidopsis in Europe grows naturally all the way from
Spain to Northern Finland. The hypothesis is then that the mechanisms that
control these developmental events might be under selective pressure to adapt to
different climates, which can then be tested by experimental growth under
multiple environments. A lot of work in evolutionary ecology of plants is
focused on understanding the molecular mechanisms involved in plant adaptation
through the control of developmental events. The assumption is then that these
mechanisms reflect adaptation to different environments and therefore hint
towards the 'why' questions. \citet{mendez-vigo_altitudinal_2011} characterise
the variation in four flowering related genes across large geographical regions
to investigate their involvement in the variation of flowering related traits
across the region. Other studies focus on seed dormancy related genes
\citep{chiang_dog1_2011} or more broadly on general characterisation of genetic
variation for multiple traits across the natural habitats of Arabidopsis
\citep{atwell_genome-wide_2010}. The developmental events are correlated since
the length of one stage affects the next, which in turn affects the next
generation. Therefore the genetic mechanisms behind two related (and
consecutive) developmental events are sometimes studied together
\citep{debieu_co-variation_2013}.

\subsubsection*{Phenology models}
The hypothesis in phenology is that plants sense and integrate environmental
signals until they reach the optimal condition for triggering the transition to
the next developmental stage. There are experimental evidence of this
integration but evidence on the mechanism are still behind although there are is
some work pointing to the clock as a possible mechanism.

Phenology models are usually phenomenological and make no attempt to represent
the mechanism for sensing and integration of the environmental effects. Instead,
they use numerical quantities to represent the progress towards the transition,
which is an integral over time of the values of environmental indicators that
affect the particular stage under consideration \citep{chuine_plant_2013}. For
example, if we know that temperature affects a developmental stage, a phenology
model might use a temperature sum to track the progress towards the transition:
$$
d(t) = \sum_{i}^{t} T(i) - T_b
$$
The development at time $t$, $d(t)$, is the sum of the temperatures at every
time unit above a base temperature $T_b$. The transition time, $t_s$, is the
time such that the development sum reaches a threshold value, $d(t_s) >
D_s$. Incorporating genetics is usually represented by adding a sensitivity value
to the above, $ d(t) = \sum_{i}^{t} k \cdot (T(i) - T_b)$ or changing the base temperature,
$T_b$. The idea is that plants adapt by changing, $k$ or $T_b$, which reflects
some molecular mechanism, to change the length of the stage and time their
transition at the optimal time depending on the natural environment.

Modelling phenology usually goes in tandem with experimental observations about
timing but the models usually stop there and they do not deal with the
evolutionary hypotheses that usually follow. These models usually appear in
ecological studies and do not consider the genetics while studies of genetic
variation of developmental timing adaptation usually do not have a modelling
component.

Furthermore, dealing with each developmental stage in isolation means, while
being simple and inducing simple 'why' hypotheses, is sometimes not enough as one
developmental stage affects the timing and length of the next and the length of
the entire generation affects the next generation and so on.

\subsubsection*{System models}
In order to be useful in evolutionary ecology other work has incorporated
genetics and fitness into the simple phenology models of the kind we have seen
above. \citet{chuine_phenology_2001}, for example, propose a model that combines
fitness with traditional phenology models to predict the distribution of a
species. The model is simple, the phenology part predicts the timing and the
fitness, given as probabilities of survival under different weather stresses,
predicts overall survival rates.

While the simple phenology models consider species mean behaviour, other work
has made the links between the genetic studies of timing mechanisms to phenology
models to understand the range of expected behaviour both within but also
between populations. For example, \citet{wilczek_effects_2009} add genetic
variation of the FLC gene (involved in control of flowering time) to a flowering
phenology model to predict variation in flowering time and vegetative season
length.

\citet{burghardt_modeling_2015} takes a more systems approach by taking an
integrative look at the whole life cycle instead of events in
isolation. Variation is also considered explicitly by using an individual-based
population model. Fitness, however, is not considered either directly as the
reproductive success of the individual in the population or indirectly as
survival rates as we have seen before. The focus is more on the timing and the
consequences of the genetic environmental interaction.

In representations in both simple and more complex evolutionary ecology models
in this domain the organism is absent. While this gives tractable
representations of complex processes and allows understanding of experimental
results, a more fundamental understanding requires the organism to fill the gaps
between the scales from genes to the population. Adding the organism to the
population level recovers a more mechanistically founded reproductive success
(fitness) as a first step towards explicitly modelling evolutionary dynamics.

% People:
% Kathleen Donohue
% Amity Wilczek
% Justin Borevitz

% although there is more work on adaptation to other things directly like
% temperature
% See Caroline Dean

\section{Languages for multi-scale biology}
% uses of models + 
% hint towards problems of representation
We have already outlined the need for formal models in the description of
biological systems. The unique advantage is their formal nature, which means
that they can be used as tools for turning hypotheses into predictions using the
formal methods that are provided by the technical language used to specify
them. Usually this involves simulating them on a computer. Their formal nature
also serves another purpose. Since they are externalisations of our knowledge
about a system, the representations can serve as documentation for our knowledge
when trying to communicate it to ourselves (for thinking) but also for others.

This view of notation or language as tool for thought and not solely as
technical object has been recognised by Kenneth Iverson who succinctly listed the
characteristics of a good notation: ``ease of expression of common constructs
in the domain, suggestivity, ability to subordinate detail, economy, and
amenability to formal proofs'' \citep{iverson2007notation}.

Modelling in biology becomes necessary when systems are complex. However, the
more complex the systems are, the harder models become to read and write. This
means that, while they might serve their first purpose as technical objects,
they fail on their second purpose as tools for thought. One reason for this
might be that the standard technical languages from Physics, like Ordinary
Differential Equations, are not adequate for the representation of the
constructive nature of the organisation of living organisms. The standard
abstraction is to ignore the organisation and describe metric properties of the
system captured with their values captured in numeric variables and their
dynamics in equations. While this abstraction of systems as numeric variables is
sometimes enough, if we really want to fundamentally understand an organism or
aspects of it, which is the case in these complex systems, we also need a
description of their constructive organisation. This was already noted, as we
have seen, by the first `structuralists' and systems theorists like Rashevsky
who pointed out the need for the representation of the 'relational aspects of
biological systems' as well as their more metric properties
\citet{rashevsky_topology_1954}. Later work focused on finding suitable theories
of organisation analogous to the equations of motion in Physics. For that
reason Rashevsky's student, Robert Rosen, was called the Newton of biology
\citep{mikulecky2001robert}.

While we appreciate the organisation is important, especially at multi-scale
representations abstraction through quantifiable properties is also needed
especially with more complex systems where we do not have the capacity or the
desire to model everything mechanistically.

For complex models like the ones we have seen above from plant, crop, and
evolutionary ecology research they are either simple enough that they are
captured with ODEs or similar formalisms or they are complex that they cannot be
easily mapped to any existing technical language in which case more pragmatic
approaches are used where the model becomes a program in a general purpose
programming language. This is problematic because the knowledge is usually
obscured in the simulation details.

The more complex models are the ones we are interested in, representing
processes like whole-plant models beyond the organism reaching to ecology and
evolution. The next chapter (Chapter~\ref{chp:relWork}) is dedicated to a more
in depth overview of existing work on languages for this domain and goes into
more details in some of the issues we highlighted above using illustrative
examples.


\section{Contributions}
The work of this thesis is concerned primarily with whole plant models that go
beyond the organism, languages for their description, and their applications. In
reference to the aims and discussion above, specifically my contributions are:
% This need to be refutable, like a contract for the thesis

\begin{itemize}
\item A notation, called \emph{Chromar}, for describing systems like the ones
  we hinted to above at the intersection of plant biology models in a natural
  environment. The language follows the structuralist tradition and is therefore
  object-based, which means that it can capture certain important aspects of the
  constructive organisation of living systems. At the same time quantifiable
  properties can also be represented allowing abstraction while further
  extensions allow an enriched expression language for these properties that
  combines regular mathematical expressions with (i) a flexible system of state
  observation based on database operations and (ii) a way of defining
  deterministically time-dependent values that can be used to describe parts of
  the systems or the natural environment that we do not wish to model
  mechanistically (Informal definition:~Chapter~\ref{chp:chromarEx}, Formal
  definition:~Chapter~\ref{chp:chromarForm}).
\item An implementation of the abstract version of Chromar as an embedded Domain
  Specific Language (EDSL) in the general-purpose programming language
  Haskell. This allows the set of expressions and types for quantifiable
  properties to come from the host language, which increases expressiveness
  while at the same time maintaining the naturalness of a domain-specific
  language (Chapter~\ref{chp:chromarImpl}).
\item An extension of the Framework Model (FM; \ssec{fm}) to the full-lifecycle
  of Arabidopsis Thaliana, including reproduction and see dormancy along with
  modifications that allow operation in realistic weather ranges. This is
  combined with full-lifecycle phenology models from ecology to provide a
  population level model that includes the organism (through the FM). This
  allows simulation experiments for determining ecological properties in
  different genotype x environment scenarios that are mechanistically founded
  and a step towards organism-centred models of evolution
  (Chapter~\ref{chp:fms}).
\item An application of the Framework Model with an engineering perspective,
  like in case of crop models. Instead of taking the traditional approach to
  engineer genomes, as we have seen, we take an alternative approach that aims
  at engineering the environment for achieving specific growth-related organism
  traits. We pose the problem as an optimal control problem and frame two
  instances of it, one where the control is direct and another one where the
  control is indirect through positioning (Chapter~\ref{chp:oc}).
\end{itemize}
