\documentclass[phd]{infthesis}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage[british]{babel}
\usepackage{microtype}
\usepackage[usenames,dvipsnames,svgnames,table]{xcolor}
\usepackage[english=british,autopunct=false]{csquotes}
\usepackage[natbib=true,style=authoryear-comp,maxbibnames=6]{biblatex}
\usepackage{graphicx}
\usepackage{textcomp}
\usepackage{wrapfig}
\usepackage{xfrac}
\usepackage{xspace}
\usepackage{mathcommon}
\usepackage[sc]{mathpazo}
\usepackage{hyperref}
\usepackage{expl3}
\usepackage{enumitem}
\usepackage{booktabs}
\usepackage{tabularx}
\usepackage{mathpartir}
\usepackage{fancyvrb}
\usepackage{inconsolata}
\usepackage{multirow}

\frenchspacing

% Bibliography
\addbibresource{fms.bib}
\bibliography{fms}

% Text
\newcommand{\ie}{i.e.\xspace}
\newcommand{\eg}{e.g.\xspace}

% Referencing
\newcommand{\chp}[1]{\S\ref{chp:#1}}
\newcommand{\sct}[1]{\S\ref{sec:#1}}
\newcommand{\ssec}[1]{\S\ref{subsec:#1}}
\newcommand{\eqn}[1]{Eq.~\ref{eq:#1}}
\newcommand{\eqns}[2]{Eq. \ref{eq:#1} and \ref{eq:#2}}
\newcommand{\lem}[1]{Lemma~\ref{lemma:#1}}
\newcommand{\lems}[2]{Lemmas \ref{lemma:#1} and \ref{lemma:#2}}
\newcommand{\thm}[1]{Th.~\ref{thm:#1}}
\newcommand{\fig}[1]{Fig.~\ref{fig:#1}}
\newcommand{\diagram}[1]{diagram~\ref{eq:#1}}
\newcommand{\app}[1]{Appendix~\ref{app:#1}}
\newcommand{\mcite}[1]{\mrcolor{gray}{#1}} % missing cite
\newcommand{\defn}[1]{Def.~\ref{def:#1}}
\newcommand{\prop}[1]{Prop.~\ref{prop:#1}}

% Math
\renewcommand{\tuple}[1]{(#1)}
\DeclareMathOperator*{\expn}{exp}
\renewcommand*{\exp}[1]{e^{\,#1}} % \mathrm{e}^{#1}}
\renewcommand{\qedsymbol}{\ensuremath{\blacksquare}}
\newcommand{\partialto}{\rightharpoonup}
\newcommand{\id}{\vec{1}} % identity function
\newcommand{\mr}[1]{\mathrm{#1}}

\newcommand{\den}[1]{\llbracket #1 \rrbracket}
\newcommand{\m}[1]{\{\!| #1 |\!\}}
\newcommand{\M}[1]{\mathcal{#1}}
\newcommand{\MS}[0]{\mathrm{M}}
\newcommand{\SQ}[0]{\mathrm{S}}
\newcommand{\s}[1]{\underline{#1}}
\newcommand{\G}[0]{\Gamma}
\newcommand{\D}[0]{\Delta}
\newcommand{\mytt}{t\!t}
\newcommand{\myff}{f\!\!f}

\newcommand{\V}{\mathrm{V}}

\newcommand{\sel}{\mathrm{sel}}
\newcommand{\fold}{\mathrm{fold}}

\newcommand{\ms}{\mathrm{ms}}


\newtheorem{mydef}{Definition}
\def\dotminus{\mathbin{\ooalign{\hss\raise1ex\hbox{.}\hss\cr
  \mathsurround=0pt$-$}}}
\setlength{\tabcolsep}{8pt}
\renewcommand{\arraystretch}{1.0}

\newcommand{\match}{m}
\newcommand{\up}[1]{\uparrow\! #1}

\newcommand{\n}{\mathrm{n}}

% Other stuff
\newcommand{\maybe}[1]{\mrcolor{gray}{#1}}
\newcommand{\todo}[1]{\mrcolor{red}{TODO: #1}}

% rules
\newcommand{\ar}[2]{\mr{#1} \! = \! {#2}}

\setlength{\tabcolsep}{8pt}
\renewcommand{\arraystretch}{1.2}

\begin{document}
\chapter{Framework Models}
As we have seen in the Introduction plant models are rarely comprehensive enough
to reconstruct the organism and evolutionary ecology models usually abstract the
organism away considering only its development as a numerical variable
(phenology).The first version of the Framework Model
\citep[FMv1;][]{chew_multiscale_2014} is an attempt at reproducing a `digital
Arabidopsis' inspired by functional-structural models that consider molecular
mechanisms as well as development. It only represents vegetative growth, which
limits it applicability to ecological studies (Section~\ref{subsec:fm}). Other
models have considered reproductive success through growth, including for
Arabidopsis. One simplified approach relates growth and fitness only to the
duration of the developmental period and not to its timing in the year, ignoring
environmental influences \citep{prusinkiewicz_evolution_2007}. On the other hand
recent ecology models have taken a more systems view considering the phenology
of the entire lifecycle of Arabidopsis plants in a natural setting and even
scaled this view to the population level using individual-based models
\citep{burghardt_modeling_2015}. However, the organism is only represented in
the usual ecological way as a numerical variable to represent its conceptual
development over time ignoring physical aspects of growth and development.

Here we combine the two approaches to present a population level model where the
organism is present. In particular in this chapter I do the following:
\begin{itemize}
\item Present \emph{FM-lite}, a simplified version of FMv1
  \citep{chew_multiscale_2014} represented in Chromar for studies that do not
    focus on circadian timing. It represents three of the four constituent
    models of FMv1 with modifications for real weather condition and without the
    clock and Flowering Time (FT) circuits and s (\ssec{fm-lite}).
  \item Present \emph{FM-life}, an extension of the FM-lite to the whole
    Arabidopsis life cycle represented in Chromar. FM-life includes FM-life to
    represent vegetative growth, and a new model of inflorescence growth
    including reproduction. As well as growth, timing of the developmental
    stages is given by phenology models, for example for seed
    dormancy. (\sct{compModels}).
\item Scale FM-life to the population level by introducing a clustering
  approximation in order to simulate FM-life tractably at the population scale
  over decades (\sct{pop}).
\item Show simulation results (ecological measure of interest) of the
  population-level model with contrasting environmental and genetic inputs to
  highlight that ecological questions can increasingly be informed by mechanstic
  understanding of growth processes \citep{millar_intracellular_2016,
    doebeli_towards_2017} (\sct{results}).
\end{itemize}


\section{Phenology models in Chromar}
\label{phenMod}
In many phenology models, the simulated plant accumulates a conceptual
development indicator in every time unit as a function of the
contributing environmental factors, until a threshold is reached for
transition to the next developmental stage. For example, in a seed type
\(\mr{Seed}(\mr{dev}:\mr{real})\), the dev attribute measures
development towards germination. A phenology rule for germination
affected by temperature and moisture, starting from dev value $d$,
could be:

\[\mr{Seed}(\ar{dev}{d}) \xrightarrow{} \mr{Seed}(\ar{dev}{d + f(temp,moist)}) \]

On average once every time unit the $\mr{dev}$ attribute of a
particular seed will be increased from the present value, \(d\), by a
function of the contributing factors $temp$ and
$moist$. Further parameters might represent how sensitive the
seed is to the environmental factors. At the threshold
$D_t$, the seed germinates to a plant and resets the
development measure to 0:

\[\mr{Seed}(\ar{dev}{d}) \xrightarrow{} \mr{Plant}(\ar{dev}{0}) \, \lbrack d > D_{t}\rbrack \]

where the expression inside the square brackets is used to indicate
conditional activity of the rule. The rule is active only when the
expression evaluates to true.

\section{FM-life: the component models}
\label{sec:compModels}
The models presented here represent the full life cycle in three stages: seed
dormancy (A, left panel, Figure~\ref{fig:overview}), vegetative growth up to
flowering (B, left panel, Figure~\ref{fig:overview}), and the reproductive stage
up to seed dispersal (C, left panel, Figure~\ref{fig:overview}). Each model (A,
B, C) includes a phenology component that represents only timing
(Section~\ref{phenology-models-in-chromar}). The vegetative and reproductive
stage models also represent biomass growth at the organ level, based on the
carbon budget of the plant. We varied genetic parameters that affect only the
timing components of A (seed dormancy, \(\psi_{i}\)) and B (floral repression
during vegetative growth, \(f_{i}\)), for comparison to
\citet{burghardt_modeling_2015}.  Each parameter value for an individual plant
can be fixed or selected probabilistically from a distribution as described
\citep{burghardt_modeling_2015}. The three models were integrated in a whole
life-cycle model of one plant (FM-life), and then extended to a population of
such plants.

\begin{figure}[tb]
  \centering
\includegraphics[width=0.9\textwidth]{figures/modelOverviewFig/modelOverview.eps}
\caption{Overview of the FM-life and population models used in this study. Left
  Overview of the models used for the growth and timing components for the three
  developmental stages: seed dormancy (A), vegetative period (B), and
  reproductive period (C) and Right Sketch of the population level model. Inputs
  to the model are the distribution of values of the two genetic parameters
  (\(f_{i}\), \(\psi_{i}\)) and weather data from some location for a number of
  years. The output is some population measure of interest, an example might be
  the total number of plants after \(k\) years.}
\label{fig:overview}
\end{figure}

\subsection{Seed dormancy model (A)}
\label{seed-dormancy-model-a}

The seed dormancy model is the Chromar version of the model of
\citet{burghardt_modeling_2015}, which is based in turn on
\citet{alvarado_hydrothermal_2002}. It represents the development of a
newly-dispersed seed from \(dev = 0\) to a threshold value, \(D_{g}\), where the
seed germinates. Above baseline levels of temperature $T_b$ and of moisture (see
below), increasing moisture and temperature speed the progress towards
germination. The additional developmental units added (hydrothermal units,
\(\mr{htu}\)) at every time unit are described by:

\[\mr{htu}( t ) = \ \left\{ \begin{matrix}
( \Psi( t ) - \Psi_{b}( t ) ) \cdot ( T( t ) - T_{b} ) \\
\ ( \Psi( t ) - \Psi_{b}( t ) )\  \cdot ( T_{o} - T_{b} ) \\
0 \\
\end{matrix} \right.\ \ \begin{matrix}
\mr{\ \ if\ \ }T_{b} < T( t ) \leq T_{o}\ \mr{and}\ \Psi_{b}( t ) < \ \Psi(t) \\
\mr{if\ \ T}( t ) > T_{o}\ \mr{and}\ \Psi_{b}( t ) < \ \Psi( t ) \\
\mr{otherwise} \\
\end{matrix}\]

where \(\Psi(t)\) (in MPa) and \(T(t)\) (in \textdegree C) give the moisture
(water potential) and temperature levels at time \(t\) respectively. The
definition distinguishes between operating in suboptimal and supraoptimal
temperatures (below or above \(T_{o}\ \mr{respectively})\ \). The baseline
moisture is used to represent the dormancy level of the seed. If \(\Psi_{b}\) is
high, the seed accumulates htu slowly for a given set of environment conditions,
whereas if \(\Psi_{b}\) is low, development is faster in the same
conditions. From an initial dormancy level, \(\psi_{i}\), seeds lose dormancy
(\(\Psi_{b}\) becomes smaller) over time at a rate $r$ that is also a function
of the environmental conditions, moisture and temperature, and represents the
observed process of after-ripening.  \(\psi_{i}\) is also used to represent the
genetic effect on dormancy, where high \(\psi_{i}\) represents stronger
dormancy.

In Chromar, the \(\mr{Seed}\) type captures information about the seed
development process:
\(\mr{Seed}(\mr{gntp}:(\mr{real},\mr{real}),\mr{dev}:\mr{real},r:real)\).  The
\(\mr{gntp}\) attribute stores the genotype of the organism, \(\psi_{i}\) (seed
dormancy level) and \(f_{i}\) (floral repression level), which is passed on to
the agents representing the later stages of development and transmitted
unchanged to the next generation.  \(\mr{dev}\) stores the cumulative
development indicator (sum of \(\mr{htu}\) up to the current timepoint), and
\(r\) stores the after-ripening up to the current timepoint. The development
rule is the following:
\begin{equation*}
\begin{split}
& \mr{Seed}(\ar{gntp}{a}, \ar{dev}{d}, \ar{r}{r}) \xrightarrow{1.0} \\ 
&\mr{Seed}(\ar{dev}{d + htu(temp, moist, f(r,temp,moist), a.\psi_i)}, 
\ar{r}{f(r, temp, moist)})
\end{split}
\end{equation*}
where \(\mr{temp}\) and \(\mr{moist}\) are fluents describing
temperature and moisture. We use the 'dot' (\(.\)) operator for
accessing the two genetic parameters of the \(\mr{gntp}\) attribute.
The following rule represents germination, starting the vegetative
stage:
\begin{equation*}
\begin{split}
& \mr{Seed}(\ar{gntp}{a},\ar{dev}{d},\ar{r}{r} ) \xrightarrow{1.0} \\
& \mr{Plant}(\ar{gntp}{a}), \mr{Root}(\ldots),\mr{Leaf}( \ldots ), \mr{Leaf}(\ldots) \, \, \lbrack d > D_{g}\rbrack
\end{split}
\end{equation*}
The abstract \(\mr{Plant}\) agent represents the plant at the vegetative stage,
along with agents for the root and the two cotyledon leaves. The initial
configuration of the organs at germination is as introduced by
\citet{chew_multiscale_2014}. Note that the genotype attribute is passed from
seed to emerged plant unchanged.

In order to illustrate the functioning of the model I show some illustrative
simulations of the $\mr{htu}$ accumulation over a year for real weather data in
two locations in Europe, Valencia in Spain and Oulu in Finland, and for two
values of the initial base moisture level, $\psi_i$. In the case of Valencia we
can see that there are two periods during the year that are favourable for
development towards germination, Spring and Autumn (Figure~\ref{fig:seedInd}
C). Development progresses faster in the low dormancy case ($\psi_i=0$) than in
the high dormancy case ($\psi_i=2.5$) as expected. The constraints of
development to two periods of the year and the binary threshold to germination
mean that sometimes if a seed misses the window for germination it has to wait
until the next favourable season (Figure~\ref{fig:seedInd} A).
\begin{figure}[tb]
  \centering
  \includegraphics[width=\textwidth]{figures/phenInd/seed.eps}
  \caption{Indicative runs of the seed dormancy model for two location and two
    genetic backgrounds (two values of the initial seed dormancy level). A Two
    runs of the seed dormancy model for yearly weather in Valencia (C) for two
    different value of $\psi_i$. The red line is the germination
    threshold. Stochasticity comes from the model representation and can
    sometimes be amplified by the binary threshold and weather constraints to
    lead to differrent germination behaviour of otherwise identical seeds (H
    case) B Same as A for weather data from Oulu (D) C Yearly weekly averages of
    moisture and temperature levels in Valencia D Weather data as in (C) for
    Oulu.  }
  \label{fig:seedInd}
\end{figure}

In the case of Oulu only summer provides favourable conditions for germination
(Figure~\ref{fig:seedInd} D). In the low dormancy case there is progress to
germination during the summer while in the high dormancy case there is no
development (Figure~\ref{fig:seedInd} B).

\subsection{Vegetative growth model (FM-lite) (B)}
\label{subsec:fm-lite}
For the vegetative stage we introduce a simplified version of FMv1
\citep{chew_multiscale_2014} for use in studies that do not focus on circadian
timing. FM-lite has three constituent models represented in Chromar with
modifications to environmental responses (see below), and without the fourth,
circadian clock and FT gene expression model of FMv1.

\subsubsection*{Timing}
\label{timing}

The timing component is the simpler flowering phenology model of
\citet{wilczek_effects_2009} rather than the augmented version in FMv1
(combination of \citet{chew_augmented_2012} with
\citet{salazar_prediction_2009}. Vegetative development extends from \(dev = 0\)
to a threshold value, \(D_{f}\), where the plant flowers. The main contributing
environmental factors are photoperiod, ambient temperature and vernalisation,
giving the modified photothermal units, \(\mr{mptu}\), at a time \(t\) as:
\[\mr{mptu}(t) = \mr{photoperiod}(t) \cdot \mr{thermal}(t) \cdot \mr{vernalisation}(t)\]
The \(\mr{vernalisation}\) term accounts for both the observed requirement for
a specific duration of exposure to cold and is also used to represent the
genetic effect on the progress towards flowering, modelled as
\(\mr{vernalisation}(t) = f(wc,f_{i})\), where $wc$ is the exposure to
cold accumulated up to \(t\) and \(f_{i}\) is the genetic parameter for the
initial floral repression, as in \citet{wilczek_effects_2009}.

In Chromar, the plant type:
\(\mr{Plant}(\mr{gntp}:(\mr{real},\mr{real}),\mr{dev}:\mr{real},\mr{wc}:\mr{real})\)
includes the genotype attributes as noted above, the development so far
(\(dev)\), and finally the accumulated winter chilling (\(wc)\). The
development rule is then:
\begin{equation*}
\begin{split}
&\mr{Plant}(\ar{gntp}{a},\ar{dev}{d}, \ar{wc}{w}) \xrightarrow{1.0} \\ 
&\mr{Plant}(\ar{dev}{d + mptu( temp, dl, a.f_{i}, w )},\ar{wc}{f(w)})
\end{split}
\end{equation*}
where $temp$ and $dl$ are fluents for temperature and day
length respectively, and $w$ is the present value of $\mr{wc}$. The transition to a
flowering plant, \(\mr{FPlant}\), follows:
$$
\mr{Plant}(\ar{gntp}{a}, \ar{dev}{d}, \ar{wc}{w}) \xrightarrow{1.0} \mr{\ FPlant}(\ar{gntp}{a}) \, \, \lbrack d > D_{f} \rbrack
$$

I again plot some illustrative runs of this model in two different locations in
Europe, Valencia and Oulu, and for two values of the initial floral repression,
$f_i$ (Figure~\ref{fig:vegInd}). In Valencia there are favourable conditions
throughout the year for progress towards flowering even in winter and the floral
repression level does not affect the development much (Figure~\ref{fig:vegInd}
A, C). In Oulu progress happens rapidly in spring when temperatures rise above 0
\textdegree C coinciding with longer photoperiods (Figure~\ref{fig:vegInd} B,
D).

\begin{figure}[tb]
  \centering
  \includegraphics[width=\textwidth]{figures/phenInd/plant.eps}
  \caption{Indicative runs of the flowering model for two locations (Valencia,
    Oulu) and two genetic backgrounds ($f_i=0.598$, $f_i=0.737$). A Two runs of
    the flowering phenology model with yearly weather for Valencia for two
    values of the initial floral repression level. The red line indicates the
    flowering threshold B Similar to (A) for Oulu C Weekly averages of
    photoperiod and temperature for a year in Valencia D Weekly averages of
    photoperiod and temperature for a year in Oulu.}
  \label{fig:vegInd}
\end{figure}



\subsubsection*{Growth}
As in FMv1 \citep{chew_multiscale_2014}, the growth component includes a carbon
budget for the plant from \citet{rasse_leaf_2006}, which in turn includes
photosynthesis rate equations based on the \citet{farquhar_biochemical_1980}
. Growth at the organ level (rosette leaves and root) is represented based on
the Greenlab model \citep{christophe_model-based_2008}. We will consider a
sucrose carbon pool (\(c\)), a starch carbon pool (\(s\)), and one pool for the
biomass of the root and each of the rosette leaves (left panel, Figure~\ref{fig:fmv1}). In
Chromar we have the following agents to store the state (amount of carbon, or
total biomass) of these pools:
\begin{figure}[p]
\centering
\resizebox{\textwidth}{!}{
\begin{tabularx}{1.1\textwidth}{c|ll} 
\toprule
\addlinespace[0.25cm]
\textsf{\emph{Flows}} & & \textsf{\emph{Rules}} \\ 
\multirow{10}[50]{*}{\includegraphics[width=0.45\textwidth]{figures/agentFlowsFig/agentFlows.pdf}} & \textsf{growth}&  $\mathrm{Leaf}(\ar{m}{m},\ar{i}{i}), \mathrm{Cell}(\ar{c}{c}) \xrightarrow{d(i, t)}$ \\
& & $\mathrm{Leaf}(\ar{m}{m+g(m)}), \mathrm{Cell}(\ar{c}{c-g(m)})$ \\ 
& & $[c-g(m) > c_b]$ \\
\addlinespace[0.2em]
& & similarly for other organs \\ 
\addlinespace[-0.5em]
& & \\
& \textsf{maint} & $\mathrm{Leaf}(\ar{m}{m}), \mathrm{Cell}(\ar{c}{c}) \xrightarrow{}$ \\
& & $\mathrm{Leaf}(\ar{m}{m}), \mathrm{Cell}(\ar{c}{c-f_m(m)})$ \\
& & $[c-f_m(m) > 0]$ \\
\addlinespace[0.2em]
& & similarly for other organs \\ 
\addlinespace[-0.5em]
& & \\
& \textsf{mobl} & $\mathrm{Leaf}(\ar{m}{m}), \mathrm{Cell}(\ar{c}{c}) \xrightarrow{}$ \\
& & $\mathrm{Leaf}(\ar{m}{m-f_t(m)}), \mathrm{Cell}(\ar{c}{c+f_t(m)})$ \\
& & $[c \leq c_b]$ \\
\addlinespace[0.2em]
& & similarly for other organs \\ 
\addlinespace[-0.5em]
& & \\
& \textsf{assim} & $\mathrm{Cell}(\ar{c}{c}, \ar{s}{s}) \xrightarrow{\phi(e(t))}$ \\
& & $\mathrm{Cell}(\ar{c}{c+k_c a_{\mathrm{ros}}}, \ar{s}{s+(1-k_c) a_{\mathrm{ros}}}) $ \\
& & $[c \leq c_b]$ \\
\addlinespace[-0.5em]
& & \\
& \textsf{sdegr} & $\mathrm{Cell}(\ar{c}{c}, \ar{s}{s}) \xrightarrow{}$ \\ 
& & $\mathrm{Cell}(\ar{c}{c+f_d(e(t)}), \ar{s}{s-f_d(e(t)})$ \\
& & $[\mathrm{night} \land s-f_d(e(t)) > 0]$ \\ 
\addlinespace[-0.5em]
& & \\
& \textsf{leaf cr} & $\mathrm{Plant}(), \mathrm{VAxis}(\ar{n}{n}) \xrightarrow{f_o(e(t))} $ \\ 
& & $\mathrm{Plant}(), \mathrm{VAxis}(\ar{n}{n+1}), \mathrm{LAxis}(\ar{i}{n+1}),$ \\
& & $\mathrm{Leaf}(\dots)$ \\
\addlinespace[0.25cm]
\bottomrule
\end{tabularx}} %}
\caption{An overview of the dynamics on FM-lite (growth component of vegetative
  stage). The dynamics take the form of flows between different reservoirs of
  carbon, here shown in a graphical way in the left panel with the corresponding
  Chromar rules on the right.  }
\label{fig:fmv1}
\end{figure}
%
\begin{itemize}
\item
  \(\mr{Cell}(c,s:\mr{real})\) An agent that stores the amount of
  carbon in the sucrose (\(c\) attribute) and starch pools (\(s\)
  attribute). The amounts are carbon totals at the whole plant level.
\item
  \(\mr{Leaf}(\mr{m}:\mr{real},\mr{i}:\mr{int})\) An agent that represents a
  rosette leaf. It has attributes for its mass (\(m\)) and its index of
  appearance (\(i\)).
\item
  \(\mr{Root}(\mr{m}:\mr{real})\) An agent that represents the root with
  an attribute for its mass (\(m\)).
\end{itemize}

For each organ we have a growth flow from the sucrose carbon pool to the mass of
the organ (\textsf{growth} rule, Figure~\ref{fig:fmv1}. The growth amount
depends on the demand function of the organ (\(d(i,t)\) rule rate function) and
its 'sink strength' (\(g(m)\)), which varies among organs. The value of the
demand function varies over time between 1 (maximum demand) and 0 (no demand) at
the end of the expansion period of the organ. The amount of carbon requested by
an organ at every time unit is \(g( m ) \cdot d(i,t)\). Depending on the
metabolic status of the whole plant (level of \(c\) pool) and the requests from
other organs, an organ will receive either the full expected amount or a portion
of it.

A flow in the opposite direction (\textsf{mobl} rule, Figure~\ref{fig:fmv1})
represents carbon mobilization from the organs if the central sucrose pool
(\(\mr{Cell}(c)\)) is reduced to a critical level. Thus each organ can be either
a net sink or source of carbon. For each organ, we also have a flow leaving the
system from the \(c\) pool for the cost of the maintenance respiration and other
processes of the organ (\textsf{maint} rule,
Figure~\ref{fig:fmv1}). Photosynthetic carbon fixation is represented by the
assimilation process (\textsf{assim} rule, Figure~\ref{fig:fmv1}). The amount of
assimilate at every time unit is the product of the photosynthesis rate, which
is a function of environmental conditions at that time step, and the projected
area of the rosette. Here we use an observable, \(a_{\mr{ros}}\), for the
effective rosette area, which is a function of the global state of the rosette
at the current time (derived from the masses of all the current leaves) and
takes into account the effect of shading, as in
\citet{christophe_model-based_2008}. The carbon partitioning function includes a
baseline partitioning to starch, then support of a target sucrose level, with
excess sucrose supporting growth and a final overflow to additional starch
production, as in \citet{rasse_leaf_2006}. At night, no photosynthesis occurs
and carbon from the starch pools flows to the sucrose pool (\textsf{sdegr} rule,
Figure~\ref{fig:fmv1}). Finally, we have the creation of new leaves, which
impacts the above processes indirectly by creating more demand for growth and
adding maintenance costs (\textsf{leaf cr} rule, Figure~\ref{fig:fmv1}). Leaves
are created by the main apical meristem (\(\mr{VAxis}\) agent) along with an
\(\mr{LAxis}\) agent that can give rise to lateral branches after flowering (see
next section).

It is interesting to note that in FMv1 carbon partitioning between processes and
organs is done deterministically whereas in our Chromar representation,
partitioning is an emergent, stochastic effect of competition for the finite
amount of sucrose carbon in the main reservoir. For example, partitioning of
carbon among organs for growth in FMv1 is done explicitly by dividing the demand
of each organ by the sum of the demands of all other organs
\(g( m ) \cdot \ \frac{d(i,\ t)}{\sum_{}^{}{d(i,t\ )}}\). In the Chromar
representation we do not have this explicit division by the global demand, which
means that the amount of carbon that an organ gets is higher at each growth
event but growth events are rarer because not all growth request are successful
(competition). In order to avoid 'starvation' of organs due to the increased
carbon amount I decreased the growth request size to $g(m)/10$ and increased the
rate of the growth rule to $10 \cdot d(i, t)$. If $c - g(m)/10 > c_b$ then the
request is succesful and the organ gets the whole amount for growth otherwise
the request is unsuccesful and the organ gets nothing. The competition recovers
the explicit partitioning of FMv1 albeit with greater variability even with the
smaller request size (see Figure~\ref{fig:comp}).

\emph{Modifications for natural conditions}\\
FMv1 was developed for lab conditions. As an initial approach to reflect
plant responses to the broader range of relevant conditions in nature,
we made the following changes:

\begin{itemize}
\item
  The rate of photosynthesis is set to 0 below 0 °C
\item
  The maintenance cost for an organ is also 0 below 0 °C
\item
  The rate of photosynthesis is affected by soil moisture through
  stomatal closure. The photosynthesis rate is affected by a stomata
  term \(f_{\mr{stom}}(moist)\), which is a simple phenomenological
  function that relates soil moisture and stomatal closure \citep{france_mathematical_1984}.
\end{itemize}

These conservative changes give a lower bound on the effects of natural
weather conditions.

\subsubsection*{Comparison of FM-lite with FMv1}
\label{comparison-of-fm-lite-with-fmv1}

In addition to the weather responses, Wilczek flowering model and emergent
carbon partitioning among organs, our model representation uses the stochastic
rule-based Chromar as opposed to the deterministic Matlab program of FMv1. In
order to compare the model representations, we simulated growth in the two
models for a fixed number of hours in lab conditions, where the modifications to
weather responses have no effect.  The two models were simulated in lab
conditions (22 °C, 12/12 light/dark cycles) for \(800\) growth hours and showed
comparable results (Figure~\ref{fig:comp}). FMv1 was simulated in Matlab while
FM-lite was simulated in the Haskell implementation of Chromar and the results
were averaged over five runs. The rosette mass results are the closest since
they represent the development of multiple Leaf agents, masking the stochastic
effects on each Leaf. The difference between the final rosette mass of the FMv1
and FM-lite (averaged over 5 runs) simulations is within 10\% of the final
rosette mass in FMv1. The stochasticity is more apparent for the root where the
growth curves are further apart. The difference between final root mass in FMv1
and FM-lite (averaged over 5 runs) is \textasciitilde{}20\% of the final root
mass in FMv1. Sucrose carbon levels are also more variable in FM-lite, since the
growth rule (removing sucrose carbon from the central pool) provides organs with
a larger amount but less frequently than the small fixed amount at every time
step in FMv1 (see previous section).

\begin{figure}[tb]
  \centering
  \includegraphics[width=0.9\textwidth]{figures/fmCompFig/fmCompLabs.jpg}
  \caption{A comparison of the original FM implementation (FMv1) with the
    adaptation used in this work (FM-lite) for 800 hours of growth. FM-lite
    simulations were performed in the Haskell implementation of Chromar and
    results were averaged over five runs. FMv1 simulations were carried out in
    the Matlab environment. A Comparison of simulated rosette mass trajectories
    between FMv1 and FM-lite B Comparison of simulated root mass trajectories
    between FMv1 and FM-lite C Comparison of simulated sucrose carbon between
    FMv1 and FM-lite and D Comparison of simulated starch carbon between FMv1
    and FM-lite.}
  \label{fig:comp}
\end{figure}

\subsection{Reproductive stage model (C)}
\label{reproductive-stage-model-c}

\subsubsection*{Timing}
\label{timing-1}

The timing component is a thermal time model from
\citet{burghardt_modeling_2015}, representing the development of the
inflorescence and seed from \(dev = 0\) at flowering, to a threshold value,
\(D_{s}\), where the plant disperses its seeds. Here there is no genetic
input. Moreover there is no input from the vegetative or growth model from the
same stage so for example a plant with a small rosette that makes very little
fruit will mature its seed at the same time as a very branchy plant with a lot
of fruit. This is a simplification and in fact throughout the FM-life model we
assume independence of timing with growth although the duration of the period
will indireclty affect growth (see indicative simulations below,
Figure~\ref{fig:reprGInd}). The thermal units that accumulate at time \(t\) are
simply the value of the temperature at \(t\) above a base temperature \(T_{b}\):

\[tu(t) = \left\{ \begin{matrix}
T(t) - T_{b} & \mr{if\ }T(t) > T_{b} \\
0 & \mr{otherwise} \\
\end{matrix} \right.\ \]

Writing into Chromar we have an
\(\mr{FPlant}(\mr{dev}:\mr{real})\) type for a flowered plant and
the following rule for its development that follows from the
\(tu\) definition above:
$$
\mr{FPlant}( \ar{dev}{d} ) \xrightarrow{} \mr{FPlant}(\ar{dev}{d + tu(temp)})
$$

Finally, the transition to seed happens when the accumulated development
reaches \(D_{s}\):
\[\mr{FPlant}( \mr{attr} = a,\mr{\ dev} = d ) \rightarrow \mr{Seed}( \mr{attr} = a,\ \mr{dev} = 0,\ r = 0 )\ \ \lbrack d > D_{s}\rbrack\]

Note that the genotype attribute of the parent plant is transferred to
the seeds unchanged.

As in the other phenology models I show some illustrative simulations of this
model in the two standard locations, Valencia and Oulu
(Figure~\ref{fig:reprInd}). In Valencia there is development throughout the year
whereas in Oulu the conditions are only favourable for only part of the year
from spring to autumn where the temperatures are above freezing.

\begin{figure}[tb]
  \centering
  \includegraphics[width=0.8\textwidth]{figures/phenInd/fplant.eps}
  \caption{Indicative runs of the seed dormancy model}
  \label{fig:reprInd}
\end{figure}

\subsubsection*{Growth}
\label{growth-1}

I developed the growth component of the reproductive stage model, which is
loosely related to the Greenlab model \citep{christophe_model-based_2008}. The
metabolic processes affecting the carbon budget of the plant are the same as in
vegetative growth but with additional organ types to represent the Arabidopsis
inflorescences. Organs appear in units (metamers) with a metamer
identifier. Each growth unit on the main axis consists of an internode (stem
between leaves), a leaf, and a lateral meristem that can give rise to a lateral
axis. We consider only the primary axis and secondary, lateral branches, thus
metamers on the lateral axis lack a further lateral meristem. All fruits on an
axis are represented on its last metamer, replacing the leaf; this metamer also
lacks a meristem.  Two indices represent metamer position: the index of the
metamer along its axis and the index of the parent metamer along the primary
axis (left panel, Figure~\ref{fig:reprModel}). We define the following new agent
types to represent this structure:

\begin{itemize}
\item
  \(\mr{INode}(i,pi:\mr{int},m:\mr{real})\) to represent the
  internode (stem between successive leaves). Attribute \(i\) is the
  temporal index of appearance in its axis (primary or lateral) and
  attribute \(\mr{pi}\) is the parent primary metamer. The cotyledons
  have indices 1 and 2 on the primary axis, for example.
\item
  \(\mr{LLeaf}(i,pi:\mr{int},m:\mr{real})\) to represent a leaf on
  the lateral axes.
\item
  \(\mr{Fruit}(i,pi:\mr{int},m:\mr{real})\) to represent a fruit
  on the axis.
\end{itemize}

\begin{figure}[p]
\centering
\resizebox{\textwidth}{!}{
\begin{tabularx}{1.05\textwidth}{c|ll} 
\toprule
\addlinespace[0.25cm]
\textsf{\emph{Structure}} & & \textsf{\emph{Rules}} \\ 
\multirow{20}[0]{*}{\includegraphics[scale=0.7]{figures/struct.pdf}} & \textsf{vgrowth}&  $\mathrm{FPlant}, \mathrm{VAxis}(\ar{n}{n}) \xrightarrow{f(e(t))}$ \\
& & $\mathrm{FPlant}, \mathrm{VAxis}(\ar{n}{n+1})$ \\
& & $\mathrm{INode}(\ar{i}{n+1}, \ar{pid}{V \, n+1}),$ \\ 
& & $\mathrm{Leaf}(\ar{i}{n+1}, \dots)$ \\
& & $[n < v_{\mathrm{max}}(n_f)]$ \\
\addlinespace[-0.15cm]
& & \\
& \textsf{vfruit} & $\mathrm{FPlant}, \mathrm{VAxis}(\ar{n}{n}) \xrightarrow{f(e(t))}$\\
& & $\mathrm{FPlant}, \mathrm{VAxis}(\ar{n}{n+1}), \mathrm{Fruit}(\ar{pi}{V \; n+1})$ \\
& & $[n = v_{\mathrm{max}}(n_f)]$ \\
\addlinespace[-0.15cm]
& & \\
& \textsf{lgrowth} & $\mathrm{FPlant}, \mathrm{LAxis}(\ar{i}{i},\ar{n}{n}), \mathrm{Fruit}(\ar{pi}{V \, i+1}) \xrightarrow{f(e(t))}$ \\
& & $\mathrm{FPlant}, \mathrm{LAxis}(\ar{n}{n+1}), \mathrm{Fruit},$\\
& & $ \mathrm{INode}(\ar{i}{n+1}, \ar{pi}{L \, i}), \mathrm{LLeaf}(\ar{i}{n+1}, \ar{pi}{L \, i})$ \\
& & $[n < l_{\mathrm{max}}(n_f, i) \wedge t_{\mathrm{del}}]$ \\
\addlinespace[-0.15cm]
& & \\
& \textsf{lfruit} & $\mathrm{FPlant}, \mathrm{LAxis}(\ar{i}{i}, \ar{n}{n}) \xrightarrow{f(e(t))}$ \\
& & $\mathrm{FPlant}, \mathrm{LAxis}(\ar{n}{n+1}), \mathrm{Fruit}(\ar{pi}{L \, i})$ \\
& & $[n = l_{\mathrm{max}}(n_f, i)]$ \\
& & \\
& & + metabolic processes (see Figure~\ref{fig:fmv1}) \\
\addlinespace[0.25cm]
\bottomrule
\end{tabularx}}
\caption{Overview of the structural part of the growth component of the
  reproductive stage model. The numbering scheme used to keep track of the
  positions of the organs in the inflorescence architecture is shown on the
  left. On the right the Chromar rules used to grow a structure like the one on
  the left panel (see main text for details).}
\label{fig:reprModel}
\end{figure}

The maximum number of inflorescence metamers on the main axis is taken to be
20\% of the number of rosette leaves at flowering time (\(n_{f}\)) and given by
\(v_{\max}(n_{f})\) \citep{pouteau_significance_2009}. The maximum number of
growth units on each lateral axis is given by \(l_{\max}(i)\), a decreasing
function of the index of the lateral axis starting from a maximum of 6 at the
axis after the cotyledons (index 3) and going to a minimum of 1 at the topmost
lateral branch \citep{mundermann_quantitative_2005}. The topmost lateral axis
can only appear with a delay after the apical fruit has appeared on the primary
axis. Each successive lateral branch going down can only start developing with a
delay after the fruit of the axis above it has appeared. The delay associated
with lateral axis growth, given in the rules by \(t_{\mr{del}}\), is a function
of the metabolic state of the plant, as described
\citep{christophe_model-based_2008}.

The new organ types have associated sink strengths and demand functions.  The
cauline leaves on the main axis contribute to the photosynthetically active area
and can shade the rosette leaves underneath them. The lateral leaves contribute
to photosynthesis without shading. Internodes and fruits do not contribute to
photosynthesis. Seeds are not directly represented, so a birth function \(b(m)\)
is required to calculate the number of seeds for a given fruit mass \emph{m} at
seed dispersal time, as described below.

I plot two indicative runs of the vegetative + reproductive stage models at two
different temperatures to show the interplay between growth and seed-set timing
(Figure~\ref{reprGInd}). At 22 \textdegree C we get two fruit agents and one
lateral branch at seed-set (1379 hours after sowing) with a total of 14 out of
25 non-senesced leaves on the main axis (Figure~\ref{reprGInd}A, B). While the
photosynthesis rate is higher at 25\textdegree C, the growth period is shorter,
which means less growth time in the exponential growth period resulting in a
smaller plant and less fruit mass (Figure~\ref{fig:reprGInd} C, D).

\begin{figure}[p]
  \centering
  \includegraphics[width=\textwidth]{figures/reprInd/reprSim.eps}
  \caption{Indicative runs of the vegetative + reproductive stage models
    (germination to seed-set). A 5-run averages of growth period indicative
    biomass traces (whole plant, main axis, lateral axes) for growth at 22
    \textdegree C (and lab conditions, light
    intensity=120$\mu \mathrm{mol} \cdot m^{-2} \cdot s^{-1}$, CO2=420ppm). B
    Sketch of the structure of the inflorescence at seed-set (1379 h) C Growth
    period simulations with same conditions as in (A) but at 25 \textdegree C. D
    Same as in (B) for growth at 25 \textdegree C.
  \label{fig:reprGInd}
\end{figure}

\section{From the lifecycle to the population}
\label{sec:pop}
The Chromar framework allows us simply to concatenate the rules of
timing and growth components of the three models above, to represent the
whole life cycle. Then given an initial state with the genetic
attributes of the plant (\(\mr{gntp}\) attribute of agents) and the
environmental conditions for a particular location, \(e(t)\), we can
simulate an entire life cycle from seed to seed. The timing components
of the model give us the timing within the year of the growth period
(vegetative + reproductive stages) and therefore the environmental
conditions that the plant is exposed to during growth. The growth
components predict growth at the individual organ level with these
environmental conditions and therefore give us the environmentally
determined seed number given by the \(b(m)\) function.



\subsection{Population level model and plotting conventions}
\label{population-level-model-and-plotting-conventions}

\begin{figure}[tb]
\centering
\includegraphics[width=\textwidth]{figures/popLevelFig/popLevel1.pdf}
\caption{The two-stage simulation of the population level model. A The
  distributions of developmental events (germination, flowering) from the
  phenology-only simulation of the population model (C) with the identification
  of two clusters representing two distinct strategies. B The 25-th, 50-th, and
  75-th percentiles of the distributions of developmental events of the two
  clusters from A. C Illustration of the phenology only simulation with b(m)=1
  (each plant makes a single seed) D Results of simulations of the growth models
  for the median dates of developmental events (B) E Illustration of the
  assignment of fruit masses to the lifecycles of the phenology-only simulation
  (C) according to their clusters. This recovers the full branching population
  process.}
\label{fig:popModel}
\end{figure}

Since FM-life estimates the number of seeds at the end of the life cycle, these
can initiate multiple independent copies of the model in the next generation. We
then have a classical evolutionary birth process, sometimes called a branching
process since it unfolds in tree-like way. The potential number of individuals
in generation \(i\), \(n_{i}\), is equal to the sum of the number of seeds
produced by the individuals in the previous generation (see Discussion). Dormant
seed never die in the model and may germinate after several years
\citep{burghardt_modeling_2015}.

Since we are using an individual-based model, \(n_{i}\) becomes computationally
prohibitive to simulate over decades of population growth. In order to overcome
this limitation, we simulated the timing (phenology) and growth components
sequentially and used conservative birth functions
\emph{b}(\emph{m}). Figure~\ref{fig:popModel} introduces the plotting
conventions for these results. The timing components were first simulated with
\(b(m) = 1\), such that each plant makes one seed, as in
\citet{burghardt_modeling_2015}. The phenological simulation results in an
unbranched sequence of developmental stage timings for each lineage (Figure~\ref{fig:popModel}
C). The simulation results for several decades typically revealed a small
number of life cycle growth strategies, from clusters of individual life
cycles. The clusters were generated using \(k\)-means clustering, where \(k\) is
chosen by visual inspection of the life cycle plots
(Figure~\ref{fig:popModel}A). Alternative clustering approaches might be an area
for future work. Figure~\ref{fig:popModel}A shows the distribution over a year
of all individual life cycles that conformed to two contrasting life cycle
strategies under environmental conditions for Valencia (see Results).  Cluster
membership depends on the dates and durations of multiple developmental
stages. This is hard to visualise, because the timing of any single
developmental stage partially overlaps among different
strategies. Figure~\ref{fig:popModel}B therefore summarises the median dates of
all three developmental transitions in each strategy, here illustrated by 1. a
summer growth strategy and 2. a winter growth strategy. In the next stage, the
growth models were simulated once per cluster, with the environmental conditions
associated with the typical timing of that cluster (median vegetative and
reproductive stages). This returns the typical biomass of organs over time,
including the fruit mass at seed dispersal (m1 for cluster 1, m2 for cluster 2;
Figure~\ref{fig:popModel}D). Finally, each life cycle is assigned the fruit mass
\emph{m} associated with its cluster, and thereby a growth-based, reproductive
success \emph{b}(\emph{m}) that evaluates to 0 in some cases. Thus, the second
stage recovers a version of the branching lineage tree, where some lineages die
out (Figure~\ref{fig:popModel}E).

Our output population measure is the total population of plants over all
lineages over all generations. For example, consider a lineage with
three generations starting with a plant with final fruit mass
\(m_{11}\). For the next generation we have \(b(m_{11})\) and then
\(b(m_{11}) \times b(m_{21})\). The population measure for that lineage
is \(1 + b(m_{11}) + b(m_{11}) \times b(m_{21})\). The population
measure for multiple lineages starting from multiple plants in the
initial population is the sum of the population measures of all the
lineages. This requires a birth function, which we use in a very simple
form, as follows:

\[b( m ) = \left\{ \begin{matrix}
1 & \mr{if\ }m > m_{0} \\
0 & \mr{otherwise} \\
\end{matrix} \right.\ \]

A plant produces one seed or none, the latter in life cycles with fruit
mass at seed dispersal \emph{m} less than a threshold \(m_{0}\). Below,
we make some conservative choices for the value of the reproductive
threshold value, \(m_{0}\), to explore the effect on the output
population measure.

Finally, we distinguish three sources of variability in the population
model: (i) weather varies between years, (ii) genetic parameters can
vary among the initial population if their values are chosen
probabilistically, and (iii) simulation results vary due to
stochasticity in the model representation.

\subsection{Weather data}
\label{weather-data}

For the phenology model simulation we used the weather data that accompanied the
\citet{burghardt_modeling_2015}, available from a Dryad repository
\citep{burghardt_data_2014}. In this dataset weather inputs over 60 years were
generated stochastically for four locations in Europe: Halle, Valencia, Norwich,
and Oulu. The weather inputs include values for temperature, moisture, and
daylength.

For the growth simulations we used weather data from the ECMWF ERA-Interim
dataset over the years 2010-2011 \citep{dee_era-interim_2011}.  A program was
used to generate hourly inputs given daily averages from the dataset for
temperature and radiation. For the soil moisture input used in the
photosynthesis rate calculation we used a daily average of soil moisture values
from the dataset and assumed that is constant throughout the day (swvl
parameters in the ERA dataset).

\section{Results}
\label{sec:results}

The population of FM-life models (see Methods) allows us to test how growth
processes that alter reproductive success affect the life history strategies of
Arabidopsis growing in different environmental conditions (location) and with
different genetic parameters in the initial population. We can therefore explore
the genotype x environment interaction, using a population measure. To
illustrate this potential, we compare simulation results for two
previously-studied locations, Valencia (Spain) and Oulu (Finland), and two
opposing combinations of genetic parameters, high seed dormancy /high floral
repression (HH) and low seed dormancy /low floral repression (LL). Within an
initial population of 100 seeds, the seed dormancy levels, \(\psi_{i}\), were
assigned probabilistically, sampling from a normal distribution with mean
\(0.0\) and standard deviation, \(1\), for the Low dormancy level (L) and mean
\(2.5\) with the same standard deviation for the High dormancy case (H). Floral
repression was fixed at either \(0.598\) for the Low level (L) and \(0.737\) for
the High level (H), values that were chosen to reflect the behaviour of natural
populations of Arabidopsis in \citet{wilczek_effects_2009}. Both parameter
choices follow \citet{burghardt_modeling_2015}. The simulation time period was
60 years and, as in \citet{burghardt_modeling_2015}, we discarded the first 15
years of the simulation to focus on stable life history strategies. A key
difference from the earlier work is that even our conservative choice of birth
function (see Methods) allows some lineages to die out.

\subsection{Valencia}
\label{valencia}
Figure~\ref{fig:valComp} shows the results of the two-stage simulation for a population
of the LL genotype in Valencia (Figure~\ref{fig:valComp}A). We identify four possible
life history strategies based on the timing (phenology) components of
the FM-life model:
\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  \emph{summer-only strategy} where the entire growth is in the summer.
  The growth period is quite short and the conditions unfavourably hot
  and dry. In the growth simulation, the rosette leaves senesce before
  the reproductive stage (blue curve). The drought effect on
  photosynthesis severely limits the carbon available for fruit mass
  (red curve).
\item
  \emph{spring strategy} where the entire growth period is in the
  spring. The growth period is only slightly longer than the
  \emph{summer-only} strategy but it falls in more favourable weather
  conditions. The rosette lifetime extends beyond flowering to support
  fruit growth, which combined with favourable weather gives high fruit
  mass.
\item
  \emph{winter-repr strategy} spans the winter/early spring period. A
  short vegetative period in the end of summer/early Autumn ends with
  flowering and a long reproductive stage over the winter/early spring.
  The rosette is senescing when favourable conditions return in early
  spring, seriously limiting fruit development.
\item
  \emph{winter-veg strategy} again spans the winter/early spring period.
  The life cycle duration is similar to strategy 3 but slightly later
  germination delays flowering until Spring. The rosette grows all
  winter, overlapping with a short reproductive stage and supporting
  high fruit mass.
\end{enumerate}

\begin{figure}[p]
\includegraphics[width=1.05\textwidth]{figures/valCompFig/valComp.eps}
\caption{Population experiments in Valencia in two different genetic backgrounds
  (LL:low dormancy, Low floral repression and HH: High dormancy, High floral
  repression). A Results of the phenology-only simulation results for the LL
  genotype. The 25-th, 50-th, and 75-th percentile of the distribution of
  developmental events are shown for each identified cluster B Illustration of
  growth stages over a year for each cluster from A according to the median time
  of the distribution of developmental events for each cluster. C Growth
  simulations over the growth period shown in panel B for each cluster.  D
  Probabilities of successive strategies. E Probabilities of successive
  strategies after eliminating life cycles with strategies 1 and 3 using a
  reproductive threshold. F, G, H, I, J Equivalently for the HH genotype in the
  same location.}
\label{fig:valComp}
\end{figure}


\begin{figure}[tb]
\includegraphics[width=\textwidth]{figures/valCompFig/valSankeys.eps}
\caption{E Probabilities of successive
  strategies after eliminating life cycles with strategies 1 and 3 using a
  reproductive threshold. F, G, H, I, J Equivalently for the HH genotype in the
  same location.}
\label{fig:valComp}
\end{figure}




Plants with life cycle strategies 2 and 4 predicted orders of magnitude more
fruit mass than plants with life cycle strategy 3 and or the least successful
strategy 1 (Figure~\ref{fig:valComp}C). This result clearly ranked the strategies available to
plants of the LL genotype, although the absolute values of the predicted biomass
are less certain (see Discussion). The 100 plants amassed 4905 potential
lifecycles over 45 years of phenological simulation (Figure~\ref{fig:valComp}A). Without a
minimum mass threshold ($m_0$) for reproduction, 66\% of potential life cycles
followed the more successful \emph{spring} and \emph{winter-veg} strategies (2
\& 4; Figure~\ref{fig:valComp}C). Figure~\ref{fig:valComp}D shows the sequential transitions between
strategies. For example, 60\% of potential plants following the successful
\emph{spring} strategy (2) disperse their seeds early enough for the next
generation to adopt the \emph{winter-veg} strategy (4), achieving two
generations per year. These transitions underlie the bimodal distribution of
life cycle times reported by \citet{burghardt_modeling_2015} for this
simulation.

Simulation of the HH genotype (Figure~\ref{fig:valComp}F, 6J) identified similar
strategies. Since the seed have longer dormancy, the population amassed
fewer potential life cycles (2954 as opposed to 4905 in the LL case;
Figure~\ref{fig:valComp}F). The growth and final fruit masses are different because of
slight variation in timing of the growth period but strategies 2 and 4
are again more successful than strategies 1 and 3 (Figure~\ref{fig:valComp}H). A higher
fraction of potential life cycles followed the successful strategies
(78\% as opposed to 66\% in the LL case; Figure~\ref{fig:valComp}G). Higher seed
dormancy reduced the germination in the summer and early autumn that led
to the less successful strategies 1 and 3, so any strategy was likely to
be followed by either strategy 2 or 4 in the next generation (Figure~\ref{fig:valComp}).

In order to calculate the population success we make two choices for the
reproduction mass threshold, \(m_{0}\), which eliminate one or both of the least
successful strategies. Choosing a value \(m_{0} = 2 \times 10^{- 5}\) g (the
mass of a single seed) eliminated the \emph{summer-only} strategy from both
genotypes, which gives a population of 1210 plants over 45 years in the LL case
(Figure~\ref{fig:valComp}C). The HH genotype allows a larger percentage of
viable life cycles but we predict fewer plants (1020) since the number of
potential life cycles was lower (Figure~\ref{fig:valComp}H). Choosing a value
\(m_{0} = 6 \times 10^{- 3}\) g left only two viable strategies, 2 and 4, for
both genotypes. Reciprocal transitions between the strategies were still
possible but \emph{winter-veg} was strongly favoured (Figure~\ref{fig:valComp}E,
J). The LL genotype predicted 240 plants in total over 45 years, compared to 360
plants for the HH genotype: G x E interaction favoured the HH genotype despite
its smaller number of potential life cycles. Thus, modelling the growth
processes not only distinguished among the potential life cycle strategies
within a genotype but also distinguished between the genotypes.

\begin{figure}[tb]
  \centering
  \includegraphics[width=\textwidth]{figures/popExps/pops.eps}
  \caption{Populations}
  \label{fig:pops}
\end{figure}

\begin{figure}[tb]
  \centering
  \includegraphics[width=\textwidth]{figures/paramDistrs/psisCorVal.eps}
  \caption{Dormancy level correlation with lineage length.}
  \label{fig:psisCorVal}
\end{figure}

\begin{figure}[tb]
  \centering
  \includegraphics[width=\textwidth]{figures/paramDistrs/psisCorOul.eps}
  \caption{Dormancy level correlation with lineage length.}
  \label{fig:psisCorOul}
\end{figure}



\subsection{Oulu}
\label{oulu}
The equivalent simulations were performed for conditions in Oulu,
Finland in the same LL and HH genetic backgrounds (Figure~\ref{fig:oulComp}). The
results indicated 3 potential life cycle strategies (Figure~\ref{fig:oulComp}A, B, E,
F):

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  \emph{summer-only strategy} where the entire life cycle occurs in the
  summer. The vegetative period is short, the rosette is very small and
  supports negligible fruit growth (Figure~\ref{fig:oulComp}C).
\item
  \emph{winter-repr strategy} where a life cycle of almost a year has a
  very short vegetative stage, followed by a long reproductive stage
  over the winter. Again, the very small rosette supports little fruit
  growth in the following Spring.
\item
  \emph{winter-veg strategy} where the plant over-winters in the
  vegetative stage. Unlike in Valencia, the rosette grows little over
  the winter. Rapid rosette growth in the following spring supports a
  substantial inflorescence and fruit development, though the predicted
  fruit mass is smaller than in Valencia.
\end{enumerate}

\begin{figure}[p]
\centering
\includegraphics[width=\textwidth]{figures/oulCompFig/oulComp.eps}
\caption{Population experiments in Oulu in two different genetic backgrounds (LL
  ? low dormancy, Low floral repression and HH-High dormancy, High floral
  repression). A Results of the phenology-only simulation results for the LL
  genotype. The 25-th, 50-th, and 75-th percentile of the distribution of
  developmental events are shown for each identified cluster B Illustration of
  growth stages over a year for each cluster according to the median time of the
  distribution of developmental events for each cluster. C Growth simulations
  over the growth period shown in panel B for each cluster.  D Probabilities of
  successive strategies. E, F, G, HJ Equivalently for the HH genotype in the
  same location.}
\label{fig:oulComp}
\end{figure}

The severe winter conditions limited the number of potential life cycles to 2361
for the LL genotype or 363 for HH. A higher proportion of HH life cycles
followed the successful \emph{winter-veg} strategy (3; 32\% against 24\% in LL;
Figure~\ref{fig:oulComp}B, G). Surprisingly, a majority of life cycles for both genotypes
followed the \emph{winter-repr} strategy (2).  Applying the reproductive
threshold mass, \(m_{0}\), eliminated one or both of strategies 1 and 2 (Figures~\ref{fig:oulComp}
C, G), suggesting a strong selective pressure for greater floral repression to
reduce the number of \emph{winter-repr} life cycles. With \(m_{0}\) =
\(2 \times 10^{- 3}\) g, the LL genotype yielded 159 plants over 45 years
compared to 53 plants for HH. All G x E combinations had actively-growing plants
at the end of the simulation. Interestingly, plants of the HH genotype had
higher average reproductive success per plant in Oulu yet the LL plants were
more successful by our population measure. The faster development of LL plants
allowed more, short lifecycles within the simulated interval (consistent with
the phenology model alone).

\section{Discussion}
\label{discussion}

We present a whole-life-cycle multi-model for growth and reproduction of
\emph{Arabidopsis thaliana}, FM-life, combining phenology models that time the
developmental stages and growth models to predict organ biomass. The simple,
FM-lite model of vegetative growth, and its extension to the reproductive stage
in FM-life, simulate broader, mechanistically-founded components of fitness at
the individual plant level compared to the phenology models alone. Most insights
from the component models naturally remain
\citep{rasse_leaf_2006,christophe_model-based_2008,
  wilczek_effects_2009,burghardt_modeling_2015}. Multi-models are helpful in
emphasising interactions. The cauline leaves in our inflorescence model, for
example, extend the duration of photosynthetic competence. As cauline leaves can
be produced 6 months later than early rosette leaves in the \emph{winter-veg}
strategy (Figure~\ref{fig:oulComp}), they remain active photosynthetic sources
\citep{earley_inflorescences_2009, leonardos_photosynthetic_2014} when the
rosette leaves are senescing. The growth models provided the fruit mass that we
used as an indicator of reproductive success, such that metabolic and
developmental processes of growth informed a more mechanistic understanding of
ecological, population dynamics over multiple generations.

The growth model allowed us to discriminate among alternative life cycle
strategies in each G x E combination, by selecting against strategies that were
compatible with the phenology models alone but had qualitatively worse
growth. In previous work, strategies with high seed dormancy in southern
Valencia and low dormancy in northern Oulu were noted to align with the
behaviour of the cognate wild populations \citep{atwell_genome-wide_2010,
  chiang_dog1_2011, mendez-vigo_altitudinal_2011, burghardt_data_2014}. In each
G x E combination, individual plants in our simulations might adopt alternative
life cycle strategies. The less-successful strategies were lethal in our model,
eliminating \textgreater{}95\% of potential lifecycles (simulated by the
phenology model alone) for the LL genotype in Valencia, for example
(Figure~\ref{fig:valComp}A, C). Thus, our results supported the observed
genotypic distinction between Valencia and Oulu, because the requirement for a
minimum fruit mass eliminated more lineages of the less-successful genotype in
each case (Figures~\ref{fig:valComp}C,H and \ref{fig:oulComp}C,G).

Our approach might appear conservative, as the binary birth function (one
seed/no seed) ignored variation in seed mass among life cycle strategies, which
might otherwise reinforce the advantage of successful strategies. The successful
genotype LL in Oulu, however, had lower reproductive success per plant than HH,
suggesting a more subtle balance of advantage. Genotypes with Low dormancy and
High floral repression (LH) are observed in far northern locations
\citep{atwell_genome-wide_2010}. We therefore simulated the LH genotype
(Figure~\ref{fig:oulLH}) . LH plants delayed flowering time enough to reduce the
frequency of potential \emph{summer-only} life cycles to 9\% compared to 15\% in
LL (Figure~\ref{fig:oulLH}B) and increased the fruit mass of the successful
\emph{winter-veg} life cycle close to the HH genotype
(Figure~\ref{fig:oulLH}C). The LH model predicted slightly higher reproductive
success overall, returning 171 life cycles (Figure~\ref{fig:oulLH}C) compared to
159 for the LL variant (Figure~\ref{fig:oulComp}C), consistent with the
observation of LH genotypes at this location.

\begin{figure}[tb]
\centering
\includegraphics[width=\textwidth]{figures/conclFigure/oulLH.eps}
\caption{Simulation results for a combined variant in Oulu (LH - low dormancy,
  high floral repression). A Results of the phenology-only simulation
  results. The 25-th, 50-th, and 75-th percentile of the distribution of
  developmental events are shown for each identified cluster B Illustration of
  growth stages over a year for each cluster according to the median time of the
  distribution of developmental events for each cluster. C Growth simulations
  over the growth period shown in panel B for each cluster.  D Probabilities of
  successive strategies.}
\label{fig:oulLH}
\end{figure}

\begin{figure}[tb]
\centering
\includegraphics[width=\textwidth]{figures/growthInvFigure/growthInv.eps}
\caption{Robustness of timing and growth simulation results.  A Sources of
  variability for the timing results. B Timing results for variable weather (60
  years) and variable genetic parameters chosen from a distribution and timing
  results for simulations with constant weather (same 1-year weather over 60
  years) and constant parameters. The timings of the strategies are very similar
  but strategy 1 life cycles have moved to strategy 3. This could be because of
  the particular year of weather data we used for the simulations. C Sources of
  variability for growth results D Growth simulation results starting from three
  different germination dates (corresponding to 25-th, 50-th, and 75-th
  percentiles of the distribution of germination times) of strategy 4 in
  Valencia LL (Figure~\ref{fig:valComp})}
\label{fig:growthInv}
\end{figure}

A limitation of our work arises from the fact that the phenology component
models of FM-life have been validated against field data
\citep{wilczek_effects_2009,burghardt_modeling_2015} whereas the growth
component models have not
\citep{rasse_leaf_2006,christophe_model-based_2008}. Biomass simulations are
inevitably sensitive to the timing of the growth period, because a longer
interval of exponential growth in good conditions rapidly changes absolute
biomass, as illustrated in Figure~\ref{fig:growthInv}D. Nonetheless, the FM-life
model predicted unreasonably high fruit mass in some cases. The binary birth
function ensured that this had no effect on our population measure. Among
possible gaps in understanding of the environmental effects on growth in natural
settings or in our representation, we repeat our previous caution
\citep{chew_multiscale_2014, chew_linking_2017} that models of nutrient balance
for Arabidopsis will be helpful. Rosette biomass in the Framework Model is
understandably sensitive to photosynthetic parameters
\citep{chew_multiscale_2014} yet these have not been validated in Arabidopsis
across the wide range of photoperiods and temperatures simulated here
\citep{walker_temperature_2013}. FM-life predicts a discretised fruit mass and
hence reproductive success for a typical representative of each life cycle
strategy, approximating an underlying, continuous distribution of fruit
mass. The accuracy of this approximation will depend on the variation within
clusters. The benefit lies in computational tractability, allowing us to
simulate differential reproductive success that is informed by understanding of
growth processes.

Our approach builds upon previous models that predict fitness and population
processes in Arabidopsis, which have focussed on developmental components of
fitness or on phenology
\citep{prusinkiewicz_evolution_2007,satake_forecasting_2013,springthorpe_flowering_2015}. Linking
these components sharpens ecological insight, by understanding the performance
of genetic variants in the environment that underlies differences in fitness
\citep[see discussions in][]{burghardt_modeling_2015, doebeli_towards_2017} and
can thus inform evolutionary hypotheses. Adding genetic variation between
generations will in future model Arabidopsis evolution explicitly, perhaps after
competing genetic variants \emph{in silico} using adaptive dynamics approaches
\citep{brannstrom_hitchhikers_2013, weise_mechanistic_2015}.  Thus, the FM-life
model offers a further tool to bridge among disciplines in plant biology,
ecology and evolution.


\singlespace

\printbibliography[heading=bibintoc]
\end{document}